### PR TITLE
Enable YJIT only x86_64 and arm64

### PR DESCRIPTION
**Pull request by hsbt**
Followed up https://github.com/rbenv/ruby-build/pull/2028

### CODE REVIEW

File: bin/ruby-build [hunk 1]
```diff
@@ -1279,6 +1279,9 @@ build_package_enable_yjit() {
     return 0
   fi
 
+  # If we aren't on x86_64 and arm64, don't enable YJIT
+  [ "$(uname -m)" = "x86_64" ] || [ "$(uname -m)" = "arm64" ] || return 0
```
Comment (k0kubun) [inline bin/ruby-build:1283]: nitpick:

```suggestion
  [ "$(uname -m)" = "x86_64" -o "$(uname -m)" = "arm64" ] || return 0
```

**Review by k0kubun — APPROVED** [review]
Thanks :+1:

**Comment by noahgibbs** [comment]
Looks good. Thank you!

File: bin/ruby-build [hunk 2]
```diff
@@ -1279,6 +1279,9 @@ build_package_enable_yjit() {
     return 0
   fi
 
+  # If we aren't on x86_64 and arm64, don't enable YJIT
+  [ "$(uname -m)" = "x86_64" -o "$(uname -m)" = "arm64" ] || return 0
```
Comment (eregon) [inline bin/ruby-build:1283]: `uname -m` is only arm64 on darwin, on linux it's aarch64 (the proper name for it :p)

Comment (eregon) [reply bin/ruby-build:1283]: Fixed in https://github.com/rbenv/ruby-build/commit/1f6b94cc9392ada220be3cd4d104b9e9ee883517

Comment (noahgibbs) [reply bin/ruby-build:1283]: YJIT should work with at least Mac M1 and Graviton Linux. We test on both regularly.

Comment (eregon) [reply bin/ruby-build:1283]: Yes: https://github.com/ruby/ruby/pull/6289/commits/b216672cfd35131d0f3890280a55b8c9c9375e36
And it's even tested on linux-aarch64: https://github.com/ruby/ruby/pull/6289/files#diff-62587956f943bb2503db7bc6dd27d0d888074a1c0ecaab3f570ad611aff0f7bb

Comment (hsbt) [reply bin/ruby-build:1283]: Thanks. I also confirm to work YJIT with AWS g series.

```
$ ruby -v --jit
ruby 3.2.0dev (2022-08-30T09:46:06Z master b98653f766) +YJIT [aarch64-linux]
```

**Pull request merged**